### ISSUE TITLE
interop_to_prod tests are being not being run (but appear green)

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#16519]
Since https://github.com/grpc/grpc/pull/16183 being merged, 
the interop tests are not being run due to an error in run_interop_tests.py script.  
Two causes:
- error in cloud_to_prod_jobspec leads to the run_interop_tests.py script terminating after the docker images are built (but no tests are run).
- for historical reasons (Jenkins), run_interop_tests.py return exitcode 0 even if the build fails (and we were relying on the generated report to show the build as red), but in this case, no report is being generated as the script just exits in an unexpected place.

See the log (e. g.):
https://source.cloud.google.com/results/invocations/76d75ac2-515e-4751-a677-74e552910270/targets/grpc%2Fcore%2Fmaster%2Flinux%2Fgrpc_interop_toprod/log

````
2018-08-30 12:04:59,352 SUCCESS: All docker images built successfully.
stty: standard input: Inappropriate ioctl for device
exception occurred:
Traceback (most recent call last):
  File "tools/run_tests/run_interop_tests.py", line 1319, in <module>
    transport_security='tls')
  File "tools/run_tests/run_interop_tests.py", line 779, in cloud_to_prod_jobspec
    transport_security_options += ['--use_tls=true']
UnboundLocalError: local variable 'transport_security_options' referenced before assignment
Removing docker image grpc_interop_node:d485bd5d-b246-4b72-a761-1797cec47f97
Removing docker image grpc_interop_nodepurejs:903020a1-35ea-421f-84ec-1660baf5f680
Removing docker image grpc_interop_php7:75ac72dc-f2bc-4a84-b5fc-def20e15e0d1
Removing docker image grpc_interop_java:c5359d2c-3021-4e34-ac05-2f44d3b24887
Removing docker image grpc_interop_python:2b78ba87-76c2-431f-a090-1a965f1aa7a2
Removing docker image grpc_interop_cxx:a79aa2be-44c7-4722-8754-112e3496bc8a
Removing docker image grpc_interop_dart:616fc201-fd8e-46f2-b187-d402ea6dfe94
Removing docker image grpc_interop_csharpcoreclr:7d5212ff-d6d4-409c-87c5-256fe0849f84
Removing docker image grpc_interop_java:39b44dcc-7c9b-4ff7-ae64-459c976cfe89
Removing docker image grpc_interop_csharp:e0db58d4-b9ce-42aa-b767-d6b4f4e46fd9
Removing docker image grpc_interop_go:4f07586c-566b-4b01-a02d-38c22f746e96
Removing docker image grpc_interop_php:8d9fa5b0-0777-4d45-b94a-85c34b5680d7
Removing docker image grpc_interop_ruby:bd5a503c-0e73-42de-8e1c-f22f0cb02813
[ID: 1293606] Build finished after 1414 secs, exit value: 0
````

**Comment by carl-mastrangelo** [comment]
At some point we should consider rewriting some of our tooling in Go.  It's pretty fast to compile and run, And has better error handling support.

**Comment by jtattermusch** [comment]
@carl-mastrangelo  I think this is not about language choice, it's about our testing scripts became pretty complex over time and also not having tests for testing scripts. Realistically, the way forward is simplification and throwing away everything that's not 100% necessary. We don't want to be in the business of building our custom testing solutions.

**Comment by carl-mastrangelo** [comment]
@jtattermusch Agreed.  I was suggesting trying out on some of the smaller scripts, since I think it would make our testing code more stable.

It has been a long time, but the Python scripting had trouble with subprocesses and hanging.

**Comment by jtattermusch** [comment]
Fixed by merging #16521 to v1.15.x and upmerging to master.

**Comment by jtattermusch** [comment]
interop_to_prod tests on master are running again:
https://source.cloud.google.com/results/invocations/77908fbd-0117-41af-b7c0-6464fcd19f83/targets/github%2Fgrpc/tests
